src/components/QrScan/QrScan.js

```diff
@@ -166,30 +138,67 @@ const QrScan = () => {
             cameraDeviceId !== '' ? { audio: false, video: { deviceId: { exact: cameraDeviceId } } }
               : { audio: false, video: { facingMode: 'environment' } }
           );
-          setCameraDeviceId(stream.getVideoTracks()[0].getSettings().deviceId);
           videoRef.current.srcObject = stream;
         }
       } catch (err) {
         throw Error(`Cannot access video: ${err.message}.`);
       }
     };
 
-    cameraPermission().then((granted) => {
-      if (granted) {
-        getUserMedia().then(() => {
-          createQrScanner(videoRef.current);
-        }, handleErrorFallback);
-      } else {
-        resetQrCodes();
-        setQrError(new Error('SCAN_CAMERA_UNAVAILABLE'));
-        history.push('/display-results');
+    /**
+   * Create QrScanner instance using video element and specify result/error conditions
+   * @param {*} videoElement HTML video element
+   */
+    const createQrScanner = async (videoElement) => {
+      if (!videoElement) {
+        if (runningQrScanner.current) {
+          qrScan.destroy();
+        }
+        return;
       }
-    });
+      qrScan = new QrScanner(
+        videoElement,
+        (results) => {
+          // runningQrScanner.current.stop();
+          setScannedData(results.data);
```

Comment:
Right after I first scan a SHC after starting up or reloading the Verifier, I see the following warning in the console:
```
Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.
    at QrScan (http://localhost:3000/static/js/main.chunk.js:4137:86)
```
coming from line 163. It looks like this may also have to do with the async qr-scanner calls. 

If this is an easy enough fix, it's probably worth addressing here, otherwise we could make a new ticket for it.

Reply:
I just added a return to the `useEffect` that should clean up `setScannedData`, so let me know if that gets rid of the error for you :thumbsup:

Reply:
It does :)